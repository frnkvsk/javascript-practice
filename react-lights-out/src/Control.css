@import url('https://fonts.googleapis.com/css2?family=Lato:ital@1&display=swap');

.Control {
  display: flex;
  align-items: center;
  justify-content: space-evenly;
  width: 455px;
  font-family: 'Lato', sans-serif;
  letter-spacing: .25rem;
  background: lightgoldenrodyellow;
  padding: 20px;
}

select, button {
  height: 30px;
  background: rgb(82, 194, 110);
  color: white;
  font-size: larger;
  margin: 10px;
}

button {
  border-radius: 10px;
  border-style: none;
}

button:hover {
  cursor: pointer;
  background: rgb(5, 204, 55);
}