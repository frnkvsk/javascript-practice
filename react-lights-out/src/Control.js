import React, {useState} from "react";
import "./Control.css";

function Control({ setDefaultProps }) {
  let [defaultProps, setProps] = useState("easy");
  const setDifficulty = level => {
    defaultProps = level;
    setProps(defaultProps.slice());
  }
  return (
    <div className="Control">      
      <div>
        <label htmlFor="difficulty">Difficulty:</label>
        <select name="difficulty" id="difficulty" onChange={e => setDifficulty(e.target.value)}>
          <option value="easy">Easy</option>
          <option value="med">Medium</option>
          <option value="hard">Hard</option>
        </select>
      </div>     
      
      <button type='submit' onClick={() => setDefaultProps(defaultProps)}>Set New Board</button>
    </div>    
  )
}

export default Control;
