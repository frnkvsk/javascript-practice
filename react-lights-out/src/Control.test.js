import React from "react";
import { render } from "@testing-library/react";
import Control from "./Control";

// smoke test
it("Control renders without crashing", () => {
  render(<Control />);
});

// snapshot test
it("matches snapshot", function() {
  const {asFragment} = render(<Control />);
  expect(asFragment()).toMatchSnapshot();
});
