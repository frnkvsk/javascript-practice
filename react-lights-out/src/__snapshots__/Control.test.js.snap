// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Control"
  >
    <div>
      <label
        for="difficulty"
      >
        Difficulty:
      </label>
      <select
        id="difficulty"
        name="difficulty"
      >
        <option
          value="easy"
        >
          Easy
        </option>
        <option
          value="med"
        >
          Medium
        </option>
        <option
          value="hard"
        >
          Hard
        </option>
      </select>
    </div>
    <button
      type="submit"
    >
      Set New Board
    </button>
  </div>
</DocumentFragment>
`;
